### PR TITLE
[Misc] Cleanup old flags, and rely on BUILD_TYPE for all features.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Description

- Rename test -> dogfood.
- Clarify the scope of each build: especially, CI/UnitTest will use dev build, and Regression/Benchmark will use dogfood build.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,106 +13,89 @@ include(cmake/util/Util.cmake)
...
-      USE_LIBXSMM
-      "Build with LIBXSMM library optimization"
-      "all"
-  )
-  dgl_feature_option(
-      USE_OPENMP
-      "Build with OpenMP"
-      "all"
-  )
+# Options for building DGL features, supported: "none", "dev", "test", "release", "all".
```
Comment (Rhett-Ying) [inline CMakeLists.txt:37]: explain a bit about all these types? what's the difference between `none` and `all`?

File: CMakeLists.txt [hunk 2]
```diff
@@ -13,106 +13,89 @@ include(cmake/util/Util.cmake)
...
-      BUILD_TYPE
-      "Type of the build: dev, test or release"
-      "dev"
-  )
-  message(STATUS "Build for ${BUILD_TYPE}")
+# Options for building DGL.
+# NOTE: do not modify this file to change option values.
+# Use bash script/build_dgl.sh -e '-DOPTION=VALUE' through command-line.
+dgl_option(
+    BUILD_TYPE
```
Comment (Rhett-Ying) [inline CMakeLists.txt:20]: we'd better make sure all builders have already turned to this new way: `BUILD_TYPE`? Or at least print a warning message if not defined.

Comment (frozenbugs) [reply CMakeLists.txt:20]: no, we just need to make sure that our release build pipeline and test pipeline are using proper BUILD_TYPE.

File: CMakeLists.txt (hunk 1 above)
Comment (frozenbugs) [reply CMakeLists.txt:37]: Done.

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**